syne_tune/optimizer/schedulers/transfer_learning/rush.py

```diff
@@ -0,0 +1,151 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+import pickle
+from typing import Dict, List, Optional
+
+import numpy as np
+
+from syne_tune.backend.trial_status import Trial
+from syne_tune.optimizer.scheduler import SchedulerDecision, TrialSuggestion
+from syne_tune.optimizer.schedulers.hyperband import HyperbandScheduler
+from syne_tune.optimizer.schedulers.transfer_learning import TransferLearningTaskEvaluations, TransferLearningScheduler
+
+
+class RUSHScheduler(TransferLearningScheduler):
+    def __init__(
+            self,
+            config_space: Dict,
+            transfer_learning_evaluations: Dict[str, TransferLearningTaskEvaluations],
+            metric: str,
+            **kwargs
+    ) -> None:
+        """
+        A transfer learning variation of Hyperband which uses previously well-performing hyperparameter configurations
+        as an initialization. Additionally, the performance of these initial configuration at the different rung levels
+        serve as a hurdle for other sampled configurations. No candidate configuration which is performing worse
+        than the best initial candidate, will be promoted.
+        Simple baseline that computes a bounding-box of the best candidate found in previous tasks to restrict the
+         search space to only good candidates. The bounding-box is obtained by restricting to the min-max of best
+         numerical hyperparameters and restricting to the set of best candidates on categorical parameters.
+
+        Reference: A resource-efficient method for repeated HPO and NAS.
+        Giovanni Zappella, David Salinas, Cédric Archambeau. AutoML workshop @ ICML 2021.
+
+        :param config_space: Configuration space for trial evaluation function.
+        :param metric: objective name to optimize, must be present in transfer learning evaluations.
+        :param transfer_learning_evaluations: dictionary from task name to offline evaluations.
+        """
+        super().__init__(config_space=config_space,
+                         transfer_learning_evaluations=transfer_learning_evaluations,
+                         metric_names=[metric])
+        points_to_evaluate = RUSHScheduler._determine_baseline_configurations(
+            config_space=config_space,
+            transfer_learning_evaluations=transfer_learning_evaluations,
+            metric=metric,
+            mode=kwargs.get('mode', 'min')
+        )
+        if 'points_to_evaluate' in kwargs:
+            points_to_evaluate += kwargs['points_to_evaluate']
+            points_to_evaluate = [dict(s) for s in set(frozenset(p.items()) for p in points_to_evaluate)]
+        kwargs['points_to_evaluate'] = points_to_evaluate
```

Comment:
Given that you do something special with this parameter, perhaps we could add it in the constructor and document the behavior since it is reasonable but somehow non obvious? (something like `"when points_to_evaluate is not None, the provided configurations are evaluated first in addition to top performing configurations from other tasks and also serve to preemptively cut underperforming configurations"`)

Reply:
It removes duplicates and preserves ordering.